bullet-featherstone/src/Base.cc

```diff
@@ -69,15 +69,38 @@ void GzMultiBody::SetJointPosForDof(
 {
   btScalar *positions =
       this->getJointPosMultiDof(_jointIndex);
-  positions[_dof] = static_cast<btScalar>(_value);
 
-  // Call setJointPosMultiDof to ensure that the link pose cache is updated.
-  this->setJointPosMultiDof(_jointIndex, positions);
+  if (this->getLink(_jointIndex).m_jointType == btMultibodyLink::eSpherical)
+  {
+    math::Quaterniond quat(positions[3], positions[0], positions[1],
+                           positions[2]);
+    math::Vector3d axis;
+    double angle;
+    quat.AxisAngle(axis, angle);
+    math::Vector3 posVec = axis * angle;
+    posVec[_dof] = _value;
+    angle = posVec.Length();
+    posVec /= angle;
```

Comment:
We should guard for angle = 0 here,
```suggestion
  if (fabs(angle) < 1e-10)
  {
    positions[0] = 0;
    positions[1] = 0;
    positions[2] = 0;
    positions[3] = 1;
  } 
  else
  {
    posVec /= angle;
   ...
   }
   this->setJointPosMultiDof(_jointIndex, positions);
```

Reply:
added check in [b1106e2](https://github.com/gazebosim/gz-physics/pull/805/commits/b1106e2823aa3d88a7fbd958d2f63986ab3e387b)